x-pack/plugins/apm/server/lib/errors/get_error_rate.ts

```diff
@@ -20,17 +23,20 @@ import { rangeFilter } from '../../../common/utils/range_filter';
 export async function getErrorRate({
   serviceName,
   groupId,
+  environment,
   setup,
 }: {
   serviceName: string;
   groupId?: string;
+  environment?: string;
   setup: Setup & SetupTimeRange & SetupUIFilters;
 }) {
   const { start, end, uiFiltersES, client, indices } = setup;
 
   const filter = [
     { term: { [SERVICE_NAME]: serviceName } },
     { range: rangeFilter(start, end) },
+    ...(environment ? [{ term: { [SERVICE_ENVIRONMENT]: environment } }] : []),
```

Comment:
It doesn't look like this supports the "no environment" option. We have the following logic elsewhere:
```ts
  if (environment === ENVIRONMENT_NOT_DEFINED) {
    return {
      bool: { must_not: { exists: { field: SERVICE_ENVIRONMENT } } },
    };
  }
  return {
    term: { [SERVICE_ENVIRONMENT]: environment },
  };
```

We might want to keep that in a helper and use it like:
```ts
[
  ...environmentFilter(environment)
]
```

Reply:
That being said: why are we handling environment separately? It's part of the `uiFiltersES` already, so doing:
```ts
[
  ...uiFiltersES
]
```

Should be enough.